File: cs/DressDiscover/DressDiscover.Api/src/DressDiscover/Api/Models/Worksheet/WorksheetExtentId.cs
namespace DressDiscover.Api.Models.Worksheet
{
    public sealed class WorksheetExtentId {
        public static WorksheetExtentId Parse(string value) {
            if (value.Length == 0) {
                throw new System.ArgumentException("id is empty", "WorksheetExtentId");
            }
            return new WorksheetExtentId(value);
        }

        private WorksheetExtentId(string value) {
            this.value = value;
        }

        public override string ToString() {
            return this.value;
        }

        private string value;
    }
}

File: ts/worksheet-knockout/src/dressdiscover/api/models/worksheet/worksheet_extent_id.ts
export class WorksheetExtentId {
    public static parse(id: string): WorksheetExtentId {
        if (id.trim().length == 0) {
            throw new RangeError("id is empty");
        }
        return new WorksheetExtentId(id);
    }

    public equals(other?: WorksheetExtentId): boolean {
        if (typeof(other) === "undefined") {
            return false;
        }
        return this._id == other._id;
    }

    public toString(): string {
        return this._id;
    }

    private constructor(id: string) {
        this._id = id;
    }

    private _id: string;
}
